src/docfx/build/moniker/MonikersProvider.cs

```diff
@@ -20,17 +21,25 @@ public MonikersProvider(Config config, MonikerRangeParser monikerRangeParser)
             _rules.Reverse();
         }
 
-        public List<string> GetMonikers(Document file)
+        public List<string> GetMonikers(Document file, JObject yamlHeader = null)
         {
-            // TODO: merge with the monikers from yaml header
+            var result = new List<string>();
             foreach (var (glob, monikers) in _rules)
             {
                 if (glob(file.FilePath))
                 {
-                    return monikers;
+                    result = monikers;
                 }
             }
```

Comment:
report  a warning if `monikerRange` hasn't been set in the config but has been set in the yaml header